utils/unittest/epoch_builder.go

```diff
@@ -14,15 +14,17 @@ import (
 type EpochBuilder struct {
```

Comment:
Regarding the changes to this component, see my comment above about including `Receipt` and `Seal` for the same ancestor block within the same payload. Unless there's a reason we should disallow this case, I suggest we handle it in `sealExtend` and simplify the changes here accordingly.

Reply:
I think this is not allowed.

In the envisaged sealing protocol, the consensus matching engine produces seals only for execution results that are already incorporated in the fork (as part of a receipt). So by the time the matching engine produces a seal, the block that contained the corresponding receipt was already added to the chain. The seal will necessarily be added to a child block.

cc @AlexHentschel @zhangchiqing

Reply:
> I think this is not allowed.

Hm, not allowed as in it is an inherently invalid payload? Or not allowed as in it is a payload our block builder implementation won't produce?

Reply:
It would be an inherently invalid payload because the seal depends on the chunk assignment of the result, which in turn depends on the ID of the block in which it was incorporated.

Reply:
see [comment above](https://github.com/onflow/flow-go/pull/179#discussion_r533715782)

Reply:
I see, too bad. Thanks for the explanation